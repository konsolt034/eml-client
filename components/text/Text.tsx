"use client";

import styled from "styled-components";

interface TextProps {
  readonly color?: string;
  readonly size?: string;
  readonly margin?: string;
}

const Text = styled.p<TextProps>`
  color: ${(props) => props.color || "#f9f9f9"};
  font-size: ${(props) => props.size || "14px"};
  ${(props) => props.margin && `margin: ${props.margin}`};
  line-height: 1.6;
`;

export default Text;
