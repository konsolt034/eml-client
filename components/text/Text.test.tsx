import { render } from "@testing-library/react";

import Text from ".";

describe("Component - Text", () => {
  it("should renders text component", () => {
    const { container } = render(<Text>{"Gift a friend this holiday"}</Text>);
    expect(container).toMatchSnapshot();
  });

  it("should render text", () => {
    const { getByText } = render(<Text>{"Hello world"}</Text>);
    expect(getByText("Hello world")).toBeTruthy;
  });
});
