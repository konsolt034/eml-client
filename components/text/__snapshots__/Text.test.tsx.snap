// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Text should renders text component 1`] = `
<div>
  <p
    class="sc-bcPKhP eFyuMb"
  >
    Gift a friend this holiday
  </p>
</div>
`;
